lib/Store.ts

```diff
@@ -127,6 +152,17 @@ export const fetchAdById = async (advertisementId: number) => {
   }
 };
 
+export const fetchAdByUserName = async (username: string) => {
+  try {
+    const { data } = await database.from("advertisement").select(
+      `id, price, negotiable_price, rating, notes,status, book (isbn, title, author, subject, year),
+        user_data (id,username,full_name, email)`
+    );
+    return data?.filter((data: any) => data.user_data.username === username);
```

Comment:
I think it would be better to filter the data from database-side (we would need to trasnfer less data)

Reply:
👍🏽

Reply:
I actually already did it by creating the view v_ad, tell me if it looks good to you: 
![image](https://github.com/ferrariofilippo/woel/assets/79911341/1b0d35a1-4160-4506-bdf1-cedcf65cb3c0)

![image](https://github.com/ferrariofilippo/woel/assets/79911341/b8280337-cb85-44f2-b009-4803538109f5)